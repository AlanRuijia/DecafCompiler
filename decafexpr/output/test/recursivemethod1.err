output prefix: /Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1
llvmas: /usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as
llc: /usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llc
cc: gcc
codegen: /Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decafexpr
stdlib: /Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decaf-stdlib.c
generating llvm code:/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decafexpr ... ok
; ModuleID = 'Test'

declare void @print_int(i32)

define i32 @test(i32) {
entry:
  %x = alloca i32
  store i32 %0, i32* %x
  %y = alloca i32
  store i32 10, i32* %y
  %x1 = load i32, i32* %x
  %y2 = load i32, i32* %y
  %AddTmp = add i32 %x1, %y2
  ret i32 %AddTmp
  ret i32 0
}

define i32 @main() {
entry:
  %x = alloca i32
  store i32 1, i32* %x
  %x1 = load i32, i32* %x
  %calltmp = call i32 @test(i32 %x1)
  call void @print_int(i32 %calltmp)
  ret i32 0
}
assembling to bitcode:/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llvm-as "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1.llvm" -o "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1.llvm.bc" ... ok
converting to native code:/usr/local/Cellar/llvm38/3.8.0/lib/llvm-3.8/bin/llc "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1.llvm.bc" -o "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1.llvm.s" ... ok
linking:gcc -o "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1.llvm.exec" "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1.llvm.s" "/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/answer/decaf-stdlib.c" ... ok
running:/Users/Sei/SFU/CMPT379/DecafCompiler/decafexpr/output/test/llvm/recursivemethod1.llvm.exec ... ok
