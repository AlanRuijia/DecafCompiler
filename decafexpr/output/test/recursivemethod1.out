Debug message: Generating program...
Debug message: Generating Extern List...
Debug message: Generating extern function...
17
Debug message: Generating package...
Debug message: Generating FieldDecls...
Debug message: Generating MethodDecl...
Generating MethodDecl Heads...test
Defined variable in line 5 : y
Codegen for Assign to y
